File: Redis/spring-redis/src/main/kotlin/com/springredis/dto/EmployeeInfo.java
package com.springredis.dto;

import org.springframework.beans.factory.annotation.Value;

import java.time.LocalDate;

public interface EmployeeInfo {
    @Value("#{target.emp_no}")
    String getEmpNo();

    @Value("#{target.salary}")
    Integer getSalary();

    @Value("#{target.from_date}")
    LocalDate getFromDate();

    @Value("#{target.full_name}")
    String getFullName();

//    Job getJob();
//
//    interface Job {
//        Long getId();
//        String getJobName();
//    }
}

File: itext/src/main/kotlin/com/itext/dto/EmployeeInfo.kt
package com.itext.dto;

import org.springframework.beans.factory.annotation.Value;

import java.time.LocalDate

interface EmployeeInfo {
    @Value("#{target.emp_no}")
    fun getEmpNo() : String

    @Value("#{target.salary}")
    fun getSalary() : Int

    @Value("#{target.from_date}")
    fun getFromDate() : LocalDate

    @Value("#{target.full_name}")
    fun getFullName() : String

}
